caffe2/core/tensor.h

```diff
@@ -32,10 +32,18 @@ class CAFFE2_API Tensor final {
     return impl_.defined();
   }
 
+  Tensor(TensorImplPtr ptr) {
+    impl_ = ptr;
+  }
```

Comment:
Ahem. If you're going to copy implementations from `at::Tensor`, we should do it right. You should `std::move()` the pointer here, or you're going to do an unnecessary extra copy.